larreco/SpacePointSolver/SpacePointSolver_module.cc

```diff
@@ -123,6 +124,7 @@ namespace reco3d {
     , fMaxIterationsReg(pset.get<int>("MaxIterationsReg"))
     , fXHitOffset(pset.get<double>("XHitOffset"))
     , fMinNHits(pset.get<unsigned int>("MinNHits"))
+    , fMaxNTriplets(pset.get<unsigned int>("MaxNTriplets", 0))
```

Comment:
The default value introduces a nasty failure mode, where a fcl typo e.g. `maxNTriplets` would silently cause the default value to be picked up.  Could this parameter be put into the fcl file instead?

Reply:
I totally agree with everything you said.  It would absolutely be better to reformulate the producer to use the confirugation validation, but that is beyond the scope of this PR.  A patch fix solution would be to not define the default value.

Reply:
My idea was to define the default value to ensure no change in behavior, regardless of how the module is configured in fcl files (i.e. whether by including the prolog or re-defining the configuration).
If it is preferred to remove the default value and move the definition to the fcl file I can definitely do that.

Reply:
Hi @cerati its a fair argument.
Just checking DUNE configs now, it looks like ProtoDUNE and FD inherit from the base fcl: 
https://github.com/DUNE/dunesw/blob/152a035560f59407f030dc767794c9cb4149d90a/fcl/common/SpacePointSolver_dune.fcl

I think triggering the CI system would confirm whether removing the default value + adding the value into FCL breaks anything

Reply:
Ok, I added a commit moving the default definition of the parameter to the fcl file. Hopefully this does not break any CI workflow.